### PR TITLE
Fix unchecked optional access in Charger.cpp

### PR DESCRIPTION
**Pull request by Pietfried**
## Describe your changes
* Fixed unchecked optional access in call_start_transaction of powermeter
* Moved ownership of session_uuid to Charger class
* Using proper arguments in call_start_transaction and call_stop_transaction of powermeter

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [ ] I have made corresponding changes to the documentation
- [x] I read the [contribution documentation](https://github.com/EVerest/EVerest/blob/main/CONTRIBUTING.md) and made sure that my changes meet its requirements

### CODE REVIEW

**Comment by Pietfried** [comment]
@golovasteek : FYI. Not sure if the changes are relevant for your platform

File: modules/EvseManager/Charger.cpp [hunk 1]
```diff
@@ -1065,6 +1066,7 @@ void Charger::start_session(bool authfirst) {
 }
 
 void Charger::stop_session() {
+    shared_context.session_uuid.clear();
```
[the remark this answers is not in the document]
Comment (Pietfried) [reply modules/EvseManager/Charger.cpp]: Right, moved it to the end of the function

File: modules/EvseManager/Charger.hpp [hunk 2]
```diff
@@ -262,6 +266,7 @@ class Charger {
         float max_current_cable{0.};
         bool transaction_active;
         bool session_active;
+        std::string session_uuid{""};
```
Comment (dorezyuk) [inline modules/EvseManager/Charger.hpp:269]: default constructor of string is already an empty string

File: modules/EvseManager/Charger.hpp [hunk 3]
```diff
@@ -112,6 +112,9 @@ class Charger {
     // Call anytime, but mostly used during charging session
     //
 
+    // Returns active session_uuid. Returns empty string if not session is active
+    std::string get_session_id();
```
Comment (dorezyuk) [inline modules/EvseManager/Charger.hpp:116]: const

File: modules/EvseManager/Charger.cpp [hunk 4]
```diff
@@ -1116,6 +1117,10 @@ void Charger::stop_transaction() {
                                       shared_context.stop_transaction_id_token);
 }
 
+std::string Charger::generate_session_uuid() {
+    return boost::uuids::to_string(boost::uuids::random_generator()());
```
Comment (dorezyuk) [inline modules/EvseManager/Charger.cpp:1121]: We can make it static

File: modules/EvseManager/evse/evse_managerImpl.cpp [hunk 5]
```diff
@@ -146,11 +137,7 @@ void evse_managerImpl::ready() {
 
     // The module code generates the reservation events and we merely publish them here
     mod->signalReservationEvent.connect([this](types::evse_manager::SessionEvent j) {
-        if (j.event == types::evse_manager::SessionEventEnum::ReservationStart) {
-            set_session_uuid();
-        }
-
-        j.uuid = session_uuid;
+        j.uuid = ""; // Reservation is not part of a session
```
Comment (dorezyuk) [inline modules/EvseManager/evse/evse_managerImpl.cpp:140]: string has a clear method https://en.cppreference.com/w/cpp/string/basic_string/clear

File: modules/EvseManager/evse/evse_managerImpl.cpp [hunk 6]
```diff
@@ -181,9 +168,7 @@ void evse_managerImpl::ready() {
             }
 
             session_started.reason = start_reason;
-
-            set_session_uuid();
-
+            const auto session_uuid = this->mod->charger->get_session_id();
```
Comment (dorezyuk) [inline modules/EvseManager/evse/evse_managerImpl.cpp:171]: previously set_session would always ensure that we have a uuid - here we only get it, which relies that we call `startSession` before - is this somehow ensured?

Comment (Pietfried) [reply modules/EvseManager/evse/evse_managerImpl.cpp:171]: This code is executed as part of the `start_session` function so its guaranteed that the session_uuid is set here

**Review by corneliusclaussen — APPROVED** [review]

**Pull request merged**